camel-core/src/main/java/org/apache/camel/processor/PollEnricher.java

```diff
@@ -217,6 +220,18 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
         Consumer delegate = consumer;
         if (consumer instanceof EventDrivenPollingConsumer) {
             delegate = ((EventDrivenPollingConsumer) consumer).getDelegateConsumer();
+
+            if (delegate instanceof GenericFileConsumer) {
```

Comment:
This wont work, we cannot change the state of the consumer as you can have concurrent pollEnrich's.

Reply:
@davsclaus, indeed, as the consumer is shared between many exchanges that are poll enriched concurrently.

As I result, I can see no option for implementing the file filtering in the GenericFileConsumer.isMatched method as its state belongs to a different context. A simple solution is to apply another isMatched check in the PollEnricher.process, when the instance of the current exchange is available. However this would constitute a enhancement of the PollEnricher  instead of reusing an existing method. What do you think?